src/tribler/core/start_core.py

```diff
@@ -177,10 +179,25 @@ def run_core(api_port, api_key, root_state_dir, parsed_args):
     logger.info('Running Core' + ' in gui_test_mode' if parsed_args.gui_test_mode else '')
     load_logger_config('tribler-core', root_state_dir)
 
+    gui_pid = GuiProcessWatcher.get_gui_pid()
```

Comment:
GUI process passes its PID to the core process by setting the `TRIBLER_GUI_PID` environment variable that the core process reads. Here I re-use this logic to determine the PID of the GUI process; the determined GUI PID is added to debug information about the Core process.